CONTRIBUTING.md

```diff
@@ -1,9 +1,203 @@
-# Hello!
-We are an open source project of [ARM mbed](www.mbed.com). Contributions via [pull request](https://github.com/armmbed/yotta/pulls), and [bug reports](https://github.com/armmbed/yotta/issues) are welcome!
+* [Contributing](#contributing)
```

Comment:
I'll review the whole thing later, but for now please don't create ToCs - the engine does that automatically and they're a lot of work to keep manually updated